### PR TITLE
feat: make bazel-lib the source-of-truth for Aspect recommended bazelrc settings

### PR DESCRIPTION
**Pull request by gregmagolan**

### CODE REVIEW

File: BUILD.bazel [hunk 1]
```diff
@@ -46,17 +46,17 @@ bzl_library(
 # write_source_files() to a git ignored subdirectory of the root
 genrule(
     name = "write_source_file_root",
-    outs = ["write_source_file-root_directory"],
-    cmd = "mkdir -p $@ && echo 'test' > $@/test.txt",
+    outs = ["write_source_file-root_directory/test.txt"],
```
Comment (gregmagolan) [inline BUILD.bazel:49]: I'm not sure why this was ever working but something about our recommended bazelrc settings now makes it fail. genrules don't work with directory outputs. they only know how to output files

**Review by alexeagle — COMMENTED** [review]
Is there an advantage to having the .aspect paths on these files?

File: .aspect/bazelrc/README.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Aspect Recommended Bazel Options
+
+The Bazel options found here are the source-of-truth for our recommended set of settings.
+They are mirrored on our docsite at https://docs.aspect.build/guides/bazelrc.
```
Comment (alexeagle) [inline .aspect/bazelrc/README.md:4]: Maybe just repeat here the typical setup with a write_source_files to copy these to your repo?

File: .aspect/bazelrc/correctness.bazelrc [hunk 3]
```diff
@@ -0,0 +1,68 @@
+# Do not upload locally executed action results to the remote cache.
+# This should be the default for local builds so local builds cannot poison the remote cache.
+# It should be flipped to `--remote_upload_local_results` on CI
+# by using `--bazelrc=.aspect/bazelrc/ci.bazelrc`.
+# Docs: https://bazel.build/reference/command-line-reference#flag--remote_upload_local_results
+build --noremote_upload_local_results
```
Comment (alexeagle) [inline .aspect/bazelrc/correctness.bazelrc:6]: Nice that we can accept PRs here unlike silo

**Comment by gregmagolan** [comment]
> Is there an advantage to having the .aspect paths on these files?

It indicates the recommended locations as well off the bat

**Review by alexeagle — APPROVED** [review]

File: .aspect/bazelrc/README.md (hunk 2 above)
Comment (gregmagolan) [reply .aspect/bazelrc/README.md:4]: Yup. Once I have that working in another repo I'll make a follow-up PR here to update this README

**Pull request merged**